File: python/2.7/factory/NYPizzaIngredientFactory.py
from PizzaIngredientFactory import *

from ThinCrustDough import *
from MarinaraSauce import *
from ReggianoCheese import *
from Garlic import *
from Onion import *
from Mushroom import *
from RedPepper import *
from SlicedPepperoni import *
from FreshClams import *

class NYPizzaIngredientFactory(PizzaIngredientFactory):
    def __init__(self):
        super(NYPizzaIngredientFactory, self).__init__()

    def createDough(self):
        return str(ThinCrustDough())

    def createSauce(self):
        return str(MarinaraSauce())

    def createCheese(self):
        return str(ReggianoCheese())

    def createVeggies(self):
        self.veggies = [str(Garlic()), str(Onion()), str(Mushroom()), str(RedPepper())]
        return self.veggies

    def createPepperoni(self):
        return str(SlicedPepperoni())

    def createClam(self):
        return str(FreshClams())

File: php/5.3/factory/com/pizza/pizzaIngredientFactory/NYPizzaIngredientFactory.php
<?
include_once("./pizzaIngredientFactory/PizzaIngredientFactory.php");

include_once("./dough/ThinCrustDough.php");
include_once("./sauce/MarinaraSauce.php");
include_once("./cheese/ReggianoCheese.php");
include_once("./veggies/Garlic.php");
include_once("./veggies/Onion.php");
include_once("./veggies/Mushroom.php");
include_once("./veggies/RedPepper.php");
include_once("./pepperoni/SlicedPepperoni.php");
include_once("./clams/FreshClams.php");

class NYPizzaIngredientFactory implements PizzaIngredientFactory {
    public function createDough() {
        return new ThinCrustDough();
    }

    public function createSauce() {
        return new MarinaraSauce();
    }

    public function createCheese() {
        return new ReggianoCheese();
    }

    public function createVeggies() {
        $this->veggies[] = new Garlic();
        $this->veggies[] = new Onion();
        $this->veggies[] = new Mushroom();
        $this->veggies[] = new RedPepper();
        return $this->veggies;
    }

    public function createPepperoni() {
        return new SlicedPepperoni();
    }

    public function createClam() {
        return new FreshClams();
    }
}
?>